homeassistant/components/wallbox/number.py

```diff
@@ -26,15 +28,70 @@
 from .entity import WallboxEntity
 
 
+def max_charging_current_value(coordinator: WallboxCoordinator) -> float:
+    """Return the maximum available value for charging current."""
+    return cast(float, coordinator.data[CHARGER_MAX_AVAILABLE_POWER_KEY])
+
+
+def min_charging_current_value(coordinator: WallboxCoordinator) -> float:
+    """Return the minimum available value for charging current."""
+    return cast(
+        float,
+        (
+            coordinator.data[CHARGER_MAX_AVAILABLE_POWER_KEY] * -1
+            if coordinator.data[CHARGER_DATA_KEY][CHARGER_PART_NUMBER_KEY][0:2]
+            in BIDIRECTIONAL_MODEL_PREFIXES
+            else 0
+        ),
+    )
+
+
+async def async_set_charging_current_value(
+    coordinator: WallboxCoordinator, value: float
+) -> None:
+    """Set the value for charging current."""
+    await coordinator.async_set_charging_current(value)
+
+
+async def async_set_energy_price_value(
+    coordinator: WallboxCoordinator, value: float
+) -> None:
+    """Set the value for energy price."""
+    await coordinator.async_set_energy_cost(value)
+
+
+@dataclass
+class WallboxNumberEntityDescriptionMixin:
+    """Load entities from different handlers."""
+
+    max_value_fn: Callable[[WallboxCoordinator], float]
+    min_value_fn: Callable[[WallboxCoordinator], float]
+    set_value_fn: Callable[[WallboxCoordinator, float], Coroutine[Any, Any, Any]]
```

Comment:
```suggestion
    set_value_fn: Callable[[WallboxCoordinator, float], Awaitable[None]]
```
Not sure if this is a right suggestion, but it looks a bit better.